### PR TITLE
adding aws secret manager

### PR DESCRIPTION
**Pull request by xuminx**
added aws secrets manager (only snowflake is using)
updated snowflake connection to key_based_authentication
updated sample settings.py and .env

### CODE REVIEW

**Comment by xuminx** [comment]
also updated redshift connection library from jaydebeapi to official redshift library, no longer need to download jar files

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+*.iml
+*/*/*/__pycache__/
+*/*/__pycache__/
+*/__pycache__/
+*.json
+*.sqlite
+*.suo
+*.pyc
+*/*.pyc
+settings.py
```
Comment (lawrence-chin) [inline .gitignore:23]: Was settings.py.sample altered to show the best way to use aws secrets?

File: functional_functions/__init__.py [hunk 2]
```diff
@@ -263,7 +271,7 @@ def get_mysql_snowflake(usr, pwd, role, warehouse_name, db_name=None, schema=Non
     return engine.connect() #, engine
 
 
-def query_snowflake(query, creds_dict=None):
+def query_snowflake(query, secret_name='fbi_snowflake_creds', creds_dict=None):
```
Comment (lawrence-chin) [inline functional_functions/__init__.py:274]: Is load_snowflake also fixed to use secrets manager?

**Review by lawrence-chin — COMMENTED** [review]
@xuminx lets discuss this asap

**Comment by xuminx** [comment]
updated load_via_aql_snowflake to use key based authentication
current, both load_via_aql_snowflake and query_snowflake will load private key inside themselves, next todo will be adding a function to load and decrypt private key

**Pull request merged**